### PR TITLE
Setup Sphinx documentation.

### PR DESCRIPTION
**Pull request by DinoBektesevic**
I didn't want to do math last Thursday, so I did documentation. Content is based on the content of the Google Doc docs outline, but was altered to fit the module layouts a bit better. This is only an example - a lot of holes and missing content to be filled in. We should all contribute to the docs together because there is so much of them.

Browse the example build [here](https://epyc.astro.washington.edu/~dinob/html/).             
I would like to point out [`psf` class](
https://epyc.astro.washington.edu/~dinob/html/search_referenceapi.html#kbmod.search.psf) in `bindings.cpp` as the example C++ class with docstrings at the moment. [This](https://epyc.astro.washington.edu/~dinob/html/run_search_referenceapi.html) is an example of not-really-completely-but-mostly-correctly documented Python module.
Note in C++ land is wrapped into 1 big module so all the classes and functions lay flat and are hard to read. In Python 1 class is 1 module so they look less cluttered. C++ should be broken in more modules and Python can be grouped up based on functionality in some places.

To build your own version: 
```
# get the required sphinx, numpydoc and theme packages
pip install -e .[docs]
# go to docs dir that has the makefile
cd docs/
# make targeting one of the available builders
make html
```
The makefile will build the `source` into `build`, optionally use the more verbose `sphinx-build source build` to target different output dir. See the list of available builders (`html`, `pdf`...) and their requirements [here](https://www.sphinx-doc.org/en/master/man/sphinx-build.html#cmdoption-sphinx-build-b)

### CODE REVIEW

File: docs/source/overview/output_files.rst [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+============
+
+KBMoD outputs a range of information about the discovered trajectories. Each filename includes a user defined suffix, allowing user to easily save and compare files from different runs. Below we use SUFFIX to indicate the user-defined suffix.
+
+The main file that most users will want to access is `results_SUFFIX.txt`. This file contains one line for each trajectory with the trajectory information (x pixel start, y pixel start, x velocity, y velocity), the number of observations seen, the estimated flux, and the estimated likelihood.
+
+The full list of output files is:
+
+* all_ps_SUFFIX.txt - All of the postage stamp images for each found trajectory.
+* filtered_likes_SUFFIX.txt - ???The likelihood of the trajectory computed only after some observations are filtered???
```
Comment (jeremykubica) [inline docs/source/overview/output_files.rst:11]: We can remove the ??? at the beginning and ending. I've confirmed that this is the case.

File: docs/source/overview/output_files.rst [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+* lc_SUFFIX.txt - ???The likelihood curves = psi/phi???
```
Comment (jeremykubica) [inline docs/source/overview/output_files.rst:12]: This should actually be "The light curves for each trajectory."

File: docs/source/overview/output_files.rst [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+* psi_SUFFIX.txt - The psi curves. Each curve contains a list of psi values corresponding to the predicted trajectory position at that time.
+* phi_SUFFIX.txt - The phi curves. Each curve contains a list of phi values corresponding to the predicted trajectory position at that time.
+* ps_SUFFIX.txt - The aggregated (mean, median, etc.) postage stamp images. One for each trajectory.
+* res_per_px_stats_SUFFIX.pkl - Per-pixel statistics. For each pixel, records statistics on the minimum likelihood, the maximum likelihood, and the number of results for trajectories starting at that pixel.
```
Comment (jeremykubica) [inline docs/source/overview/output_files.rst:16]: We can delete this entry as I recently removed this file.

**Review by jeremykubica — APPROVED** [review]
This looks really good. Thank you for putting it together. I made a few small comments (reflecting changes since the draft documentation was written).

File: docs/source/overview/output_files.rst (hunk 3 above)
Comment (DinoBektesevic) [reply docs/source/overview/output_files.rst:16]: You specify a single entry but commented on multiple lines - I've assumed you meant `res_per_px_stats_SUFFIX.pkl` but maybe that was incorrect?

Comment (jeremykubica) [reply docs/source/overview/output_files.rst]: Yeah. I'm not sure how I selected multiple lines. I just meant res_per_px_stats_SUFFIX.pkl

**Comment by DinoBektesevic** [comment]
I'm making this a full PR because it at least lets us build the docs locally until we figure out some kind of infrastructure to rebuild and host them automatically.

**Review by jeremykubica — APPROVED** [review]

**Pull request merged**